tests/unit/lms/views/predicates/_lti_launch_test.py

```diff
@@ -51,6 +52,84 @@ def test_when_request_params_are_missing(self, pyramid_request, value, expected)
         assert result is expected
 
 
+class TestBlackboardCopied:
+    @pytest.mark.parametrize(
+        "params",
+        [
+            # If there's already a ModuleItemConfiguration for the resource_link_id
+            # then blackboard_copied is False.
+            dict(
+                resource_link_id_exists=True,
+                resource_link_id_history_exists=True,
+                expected_result=False,
+            ),
+            dict(
+                resource_link_id_exists=True,
+                resource_link_id_history_exists=False,
+                expected_result=False,
+            ),
+            # If there isn't a ModuleItemConfiguration for the
+            # resource_link_id_history then blackboard_copied is False.
+            dict(
+                resource_link_id_exists=False,
+                resource_link_id_history_exists=False,
+                expected_result=False,
+            ),
+            # If there's a ModuleItemConfiguration for resource_link_id_history but
+            # not yet one for resource_link_id then blackboard_copied is True.
+            dict(
+                resource_link_id_exists=False,
+                resource_link_id_history_exists=True,
+                expected_result=True,
+            ),
+        ],
+    )
+    def test_it(self, params, pyramid_request):
+        tool_consumer_instance_guid = pyramid_request.params[
+            "tool_consumer_instance_guid"
+        ]
+        resource_link_id = pyramid_request.params["resource_link_id"]
+        resource_link_id_history = pyramid_request.params["resource_link_id_history"]
+
+        if params["resource_link_id_exists"]:
+            factories.ModuleItemConfiguration(
+                tool_consumer_instance_guid=tool_consumer_instance_guid,
+                resource_link_id=resource_link_id,
+            )
+
+        if params["resource_link_id_history_exists"]:
+            factories.ModuleItemConfiguration(
+                tool_consumer_instance_guid=tool_consumer_instance_guid,
+                resource_link_id=resource_link_id_history,
+            )
+
+        expected_result = params["expected_result"]
+
+        predicate = BlackboardCopied(True, mock.sentinel.config)
+        assert predicate(mock.sentinel.context, pyramid_request) == expected_result
+
+        predicate = BlackboardCopied(False, mock.sentinel.config)
+        assert predicate(mock.sentinel.context, pyramid_request) == (
+            not expected_result
+        )
+
+    def test_with_request_params_missing(self, pyramid_request):
+        pyramid_request.params = {}
+
+        predicate = BlackboardCopied(True, mock.sentinel.config)
+        assert not predicate(mock.sentinel.context, pyramid_request)
+
+        predicate = BlackboardCopied(False, mock.sentinel.config)
+        assert predicate(mock.sentinel.context, pyramid_request)
```

Comment:
What's the first truthy argument here?

Reply:
It's the value that was passed to the predicate: `@view_config(blackboard_copied=True)`. You can also do `@view_config(blackboard_copied=False)`